lighthouse-core/lib/i18n/README.md

```diff
@@ -0,0 +1,258 @@
+
+*   Inline code block.
+
+    To format some text as code it should be contained in `backticks`. Any text within the backticks will not be translated. This should be used whenever code is non-translatable. Such as HTML tags or snippets of code. Also note that there is no escape character for using backticks as part of the string, so ONLY use backticks to define code blocks.
+
+    Example usage:
+
+    ```javascript
+    const UIStrings = {
+      title: 'Document has a `<title>` element',
+    };
+    ```
+
+*   Link.
+
+    To convert a section of text into a link to another URL, enclose the text itself in [brackets] and then immediately include a link after it in (parentheses). Note that `[link text] (https://...)` is NOT VALID because of the space and will not be converted to a link.
+
+    Example usage:
+
+    ```javascript
+    const UIStrings = {
+      description: 'The value of ... [Learn More](https://google.com/)',
+    };
+    ```
+
+### ICU Replacement Syntax
+
+ICU syntax is used throughout Lighthouse strings, and they are specified directly in UIStrings. They follow 1 of 2 flavors.
+
+*   Direct replacement
+
+    This is simply a direct replacement of text into a string. Often used for proper nouns, code, or other text that is dynamic and added at runtime, but requires no additional formatting.
+
+    ICU replacements must use a JSDoc type syntax to specify an example for direct ICU replacements.
+
+    *   To specify the description, use
+
+        `@description <description>`
+
+    *   To specify an example for an ICU replacement, use
+
+        `@example {<ICU variable name>} <example for ICU replacement>`
+
+    ```javascript
+    const UIStrings = {
+        /**
+         * @description Error message explaining ...
+         * @example {errorCode} NO_SPEEDLINE_FRAMES
+         */
+        didntCollectScreenshots: `Chrome didn't .... ({errorCode})`,
+    };
+    ```
+
+*   Complex replacement
+
+    When more complex numerical ICU replacement is needed the syntax is mostly the same as direct replacement. This is often used when replcaing times, or percentages.
+
+    Note: these complex ICU formats are automatically given example values based on their ICU format as specified in `collect-strings.js`.  Therefore a normal description string can be used.
+
+    ```javascript
+    const UIStrings = {
+        /** Description of display value. */
+        displayValueText: 'Interactive at {timeInMs, number, seconds} s',
+    };
+    ```
+
+TODO(exterkamp): explain all the comments and where they go/what they become.
+
+TODO(exterkamp): explain why we can't use some ICU like number formatting.
+
+TODO(exterkamp): example of plural and ordinal.
+
+### The pipeline
+
+The translation pipeline has 2 distinct stages, the collection and translation is done at build time, and the replacement is done at runtime.
+
+#### Translation Pipeline (build time)
```

Comment:
I have a proposal for establishing some names that should simplify things.

1. **"Lighthouse Locale Format"** (LHL) is a thing. It's the ICU-friendly format that is used to author `UIStrings` and is seen in the locale files in `i18n/locales/*.json`
1. Adopt the term **"CTC format"** to refer to the json format that's currently known as "pre-locale", "messages.json", and "chrome_app". We need one easy name that mostly communicates its the format that Google TC (translation console) consumes. (CTC could mean _Chrome Translation Console_, but it doesn't need to.. it's just a distinct name). 
1. Name any files in CTC format as **`*.ctc.json`**. (These remain .gitignore'd)
1. `yarn i18n` (renamed from i18n:collect-strings) does these things:
   1. collect strings from core
   1. collect strings from stack packs
   1. writes `en-US.ctc.json` to disk to prepare for the TC import
   1. bakes en-US & en-XL and writes their locale files (enabling instant XL feedback)
1. The `export-tc-dump-to-github.sh` that we use on the google-side runs a `bake-ctc-to-lhl.js` script that lives in the LH repo, and writes/updates the `i18n/locales/*.json` files. No extra npm script run needed.


So the overall user flows look like this:

* LH contributor makes any changes to strings. They run `yarn i18n`
  * They can now use the en-XL locale to verify things work as intended.
* Googler is ready to kick off the TC pipeline again. 
  * They run `yarn i18n` and then on the google-side: `import-source-from-github.sh`, submit CL. Wait ~two weeks..
  * They run `export-tc-dump-to-github.sh`, and opens a LH PR.

Reply:
And now.. MY ASCII MASTERPIECE.

```
 Source files:                                         Locale files:
+---------------------------+                         +----------------------------------------------
|                           ++                        | lighthouse-core/lib/i18n/locales/en-US.json |
| const UIStrings = { ... };|-+                 +---> | lighthouse-core/lib/i18n/locales/en-XL.json |
|                           |-|                 |     +----------------------------------------------+
+-----------------------------|                 |     |                                             ||
 +----------------------------|                 |     | lighthouse-core/lib/i18n/locales/*.json     |-<+
  +---------------------------+                 |     |                                             || |
                           |                    |     +----------------------------------------------| |
                           |                    |      +---------------------------------------------+ |
              $ yarn i18n  +--------------------+                                                      |
                           |                                                                           |
                           v                          ▐                       ▐    +---------------+   |
              +------------+------+                   ▐   Google TC Pipeline  ▐ +->|  *.ctc.json   |---+
              |  en-US.ctc.json   |  +--------------> ▐      (~2 weeks)       ▐    +---------------+
              +-------------------+  $ g3/import….sh  ▐                       ▐ $ g3/export….sh

```

Reply:
You have no idea how excited I am that we will have a format called `CTC` 😍😃😍😃😍😃


In all seriousness, this sounds great to me 👍

Reply:
Pure ASCII gold.

Reply:
Done!